### PR TITLE
DP-2330 Data Sources documentation

### PR DESCRIPTION
**Pull request by Mara3l**
Please do not merge this Pull Request.

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by hkad98** [comment]
Let's have a sync regarding the docstrings.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/docs/hugo/docs/content/en/docs/data/data-source/load_and_put_declarative_data_sources.md [hunk 1]
```diff
@@ -11,6 +11,23 @@ superheading: "catalog_data_source."
...
+{{% parameters-block  title="Parameters"%}}
+
+{{< parameter p_name="layout_root_path" p_type="CatalogDeclarativeDataSources" >}}
+Path to the root of the layout directory. Defaults to Path.cwd().
+{{< /parameter >}}
+{{< parameter p_name="credentials_path" p_type="Optional[Path]" >}}
+Path to the credentials. Defaults to Path.cwd().
+{{< /parameter >}}
+{{< parameter p_name="test_data_sources" p_type="Optional[Bool]" >}}
+TODO hkad98. Defaults to False.
```
Comment (hkad98) [inline gooddata-sdk/docs/hugo/docs/content/en/docs/data/data-source/load_and_put_declarative_data_sources.md:23]: If True, the connection to the data source is tested.

File: gooddata-sdk/docs/hugo/docs/content/en/docs/data/data-source/put_declarative_data_sources.md [hunk 2]
```diff
@@ -11,6 +11,24 @@ superheading: "catalog_data_source."
...
+{{< parameter p_name="declarative_data_sources" p_type="CatalogDeclarativeDataSources" >}}
+Declarative Data Source object. Can be retrieved by get_declarative_data_sources.
+{{< /parameter >}}
+
+{{< parameter p_name="credentials_path" p_type="Optional[Path]" >}}
+Path to the Credentials. Optional, defaults to None.
+{{< /parameter >}}
+
+{{< parameter p_name="test_data_sources" p_type="Optional[bool]" >}}
+TODO hkad98. Defaults to False.
```
Comment (hkad98) [inline gooddata-sdk/docs/hugo/docs/content/en/docs/data/data-source/put_declarative_data_sources.md:25]: If True, the connection of data sources is tested.

File: gooddata-sdk/docs/hugo/docs/content/en/docs/data/data-source/scan_schemata.md [hunk 3]
```diff
@@ -13,6 +13,22 @@ Returns *list[str]*.
...
+{{< parameter p_name="data_source_id" p_type="string" >}}
+Data Source identification string. e.g. "demo"
+{{< /parameter >}}
+
+{{% /parameters-block %}}
+
+{{% parameters-block title="Returns"%}}
+
+{{< parameter p_type="list[string]" >}}
+List of configurable data source entities. TODO hkad98 (specify better pls)
```
Comment (hkad98) [inline gooddata-sdk/docs/hugo/docs/content/en/docs/data/data-source/scan_schemata.md:27]: List of schema names for the given data source specified by its id.

File: gooddata-sdk/gooddata_sdk/catalog/data_source/service.py [hunk 4]
```diff
@@ -159,6 +399,25 @@ def scan_data_source(
...
+            data_source_id (str):
+                Data Source identification string. e.g. "demo"
+            scan_request (CatalogScanModelRequest, optional):
+                Options for the Scan Request. Defaults to CatalogScanModelRequest().
+            report_warnings (bool, optional):
+                Switch to turn on warnings. Defaults to False.
+
+        Returns:
+            CatalogScanResultPdm:
+                TODO hkad98
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/data_source/service.py:419]: An instance of CatalogScanResultPdm. Containing pdm itself and a list of warnings that occurred during scanning.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by hkad98 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR198

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul/31/ : SUCCESS in 1m 30s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/181/ : SUCCESS in 2m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/184/ : SUCCESS in 26s

**Pull request merged**